src/Microsoft.TemplateEngine.Abstractions/Parameters/ParameterSetData.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace Microsoft.TemplateEngine.Abstractions.Parameters;
+
+public class ParameterSetData : ParametersDefinition, IParameterSetData
+{
+    public ParameterSetData(IParametersDefinition parameters, IReadOnlyList<ParameterData> parameterData)
+        : base(parameters.AsReadonlyDictionary())
+    {
+        ParametersData = parameterData.ToDictionary(d => d.ParameterDefinition, d => d);
+    }
+
+#pragma warning disable RS0027 // Public API with optional parameter(s) should have the most parameters amongst its public overloads
+    public ParameterSetData(ITemplateInfo templateInfo, IReadOnlyDictionary<string, string?>? inputParameters = null)
+#pragma warning restore RS0027 // Public API with optional parameter(s) should have the most parameters amongst its public overloads
+        : this(templateInfo, inputParameters?.ToDictionary(p => p.Key, p => (object?)p.Value))
+    { }
+
+    public ParameterSetData(ITemplateInfo templateInfo, IReadOnlyDictionary<string, object?>? inputParameters)
+        : base(templateInfo.ParametersDefinition)
+    {
+        ParametersData = templateInfo.ParametersDefinition.ToDictionary(p => p, p =>
+        {
+            object? value = null;
+            bool isSet = inputParameters != null && inputParameters.TryGetValue(p.Name, out value);
+            return new ParameterData(p, value, isSet ? DataSource.Host : DataSource.NoSource);
+        });
+    }
+
+    public IReadOnlyDictionary<ITemplateParameter, ParameterData> ParametersData { get; }
+
+    [Obsolete("IParameterSet should not be used - it is replaced with IParameterSetData", false)]
+    public static IParameterSetData FromLegacyParameterSet(IParameterSet parameterSet)
```

Comment:
It can - but this helper helps the consumers of new API bridge to it from their old data models - hence it's defined in Abstractions (so that it doesn't need to be reinvented by users). It's marked as obsolete so that it is clear it should not be used in cases where the logic is being written from scratch and hence new data models can be used